scene/2d/navigation_agent_2d.cpp

```diff
@@ -328,6 +336,22 @@ bool NavigationAgent2D::get_navigation_layer_value(int p_layer_number) const {
 	return get_navigation_layers() & (1 << (p_layer_number - 1));
 }
 
+void NavigationAgent2D::set_pathfinding_algorithm(const NavigationPathQueryParameters2D::PathfindingAlgorithm p_pathfinding_algorithm) {
+	if (pathfinding_algorithm == p_pathfinding_algorithm) {
+		return;
+	}
+
+	navigation_query->set_pathfinding_algorithm(p_pathfinding_algorithm);
```

Comment:
Need to update the local field on the node as well.

Applies to the other setters in this PR as well.